src/download.rs

```diff
@@ -401,6 +463,25 @@ mod error {
     #[derive(Debug, Snafu)]
     #[snafu(visibility = "pub(super)")]
     pub(super) enum Error {
+        #[snafu(display("Failed to read metadata for '{}': {}", path.display(), source))]
+        ReadFileMetadata {
+            path: PathBuf,
+            source: std::io::Error,
+        },
+
+        #[snafu(display("Failed to get block device size: {}", source))]
+        GetBlockDeviceSize { source: nix::Error },
+
+        #[snafu(display("Got invalid block device size: {}", result))]
+        InvalidBlockDeviceSize { result: i32 },
+
+        #[snafu(display(
+            "Block device too small: block device size {} GiB, needed at least {} GiB",
+            block_device_size,
+            needed
+        ))]
+        BadBlockDeviceSize { block_device_size: i64, needed: i64 },
```

Comment:
I think I'd prefer to make this error type either more specific, like `BlockDeviceTooSmall`.  Currently the error name is generic, but is actually a "too small" error.

Reply:
Thanks Zac, for reviewing.  I have made the change.